src/App.js

```diff
@@ -37,28 +37,60 @@ const NoLogin = () => {
  * Main App component. Wrap the main react-router components together with
  * the various dialogs and error messages that may be needed.
  */
-const App = ({ history, config, appReady }) => {
-  if (!config.get('loginToken')) { // login is missing
+/* const App = ({ history, config, appReady }) => {
+
+} */
+
+class App extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      hasError: false,
+    }
+  }
+
+  componentDidCatch (error, info) {
+    console.error(error, info)
+    this.setState({ hasError: true, error: error, errorInfo: info })
+  }
+
+  render () {
+    const { history, config, appReady } = this.props
+    const { hasError, error } = this.state
+
+    if (!config.get('loginToken')) { // login is missing
+      return (
+        <Grid fluid>
+          <NoLogin />
+        </Grid>
+      )
+    }
+
+    if (hasError) {
+      return (
+        <div>
+          <h1>Error</h1>
+          <p>{error.message}</p>
+          <pre>{error.stack}</pre>
+        </div>
```

Comment:
Pull this `div` out to another component like `NoLogin`.  Then it should be easier to quickly find the place to make any changes to the presentation and those changes won't touch the `App` component.

It'll need a bunch of (static component) polish in the near future.

Reply:
done